router/delivery/http/router_handler.go

```diff
@@ -59,13 +63,98 @@ func NewRouterHandler(e *echo.Echo, us mvc.RouterUsecase, tu mvc.TokensUsecase,
 	e.GET(formatRouterResource("/state"), handler.GetRouterState)
 }
 
+// GetQuoteRequest represents swap quote request for the /router/quote endpoint.
+type GetQuoteRequest struct {
```

Comment:
Idea is to reduce HTTP handler logic, and try to make self contained requests that are capable to: validate them selves, unmarshal data from HTTP. I was wondering whether such approach would be acceptable?

Reply:
This is a good idea, and I'm happy to experiment. However, if we follow this pattern, I would love for us to define a plan to switch the remaining APIs onto the same as a follow-up and have tasks cut for that before merging 🙏 .

I think we would benefit from a general `types` package per module (i.e `router`, `pools`, `tokens`).  We could put the requests into such a package. Later on, we could also move on module-specific errors into the same location.

Such `types` approach would make us consistent with the chain logic where [every module](https://github.com/osmosis-labs/osmosis/tree/main/x/concentrated-liquidity) has a `types` package.

Reply:
Thanks for sharing your ideas! I think it's a way to go then, I have moved GetQuoteRequest to `types` package and created [DATA-287](https://linear.app/osmosis/issue/DATA-287/encapsulate-requests) to address rest of the endpoints and requests.